portal/src/graphql/adminapi/AddRolesScreen.tsx

```diff
@@ -0,0 +1,185 @@
+    const errors = validateRole({
+      key: rawState.roleKey,
+      name: rawState.roleName,
+      description: rawState.roleDescription,
+    });
+    if (errors.length > 0) {
+      return makeLocalValidationError(errors);
+    }
+    return null;
+  }, []);
+
+  const submit = useCallback(
+    async (rawState: FormState) => {
+      const sanitizedRole = sanitizeRole({
+        key: rawState.roleKey,
+        name: rawState.roleName,
+        description: rawState.roleDescription,
+      });
+      return createRole(
+        sanitizedRole.key,
+        sanitizedRole.name,
+        sanitizedRole.description
+      );
+    },
+    [createRole]
+  );
+
+  const form = useSimpleForm({
+    stateMode:
+      "ConstantInitialStateAndResetCurrentStatetoInitialStateAfterSave",
+    defaultState,
+    submit,
+    validate,
+  });
+
+  const { state: formState, setState: setFormState } = form;
+
+  const breadcrumbs = useMemo<BreadcrumbItem[]>(() => {
+    return [
+      {
+        to: "~/user-management/roles",
+        label: <FormattedMessage id="RolesScreen.title" />,
+      },
+      { to: ".", label: <FormattedMessage id="AddRolesScreen.title" /> },
+    ];
+  }, []);
+
+  const onFormStateChangeCallbacks = useMemo(() => {
+    const createCallback = (key: keyof FormState) => {
+      return (e: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+        const newValue = e.currentTarget.value;
+        setFormState((prev) => {
+          return { ...prev, [key]: newValue };
+        });
+      };
+    };
+    return {
+      roleKey: createCallback("roleKey"),
+      roleName: createCallback("roleName"),
+      roleDescription: createCallback("roleDescription"),
+    };
+  }, [setFormState]);
+
+  return (
+    <FormContainerBase form={form} afterSave={afterSave}>
+      <RoleAndGroupsFormLayout
+        breadcrumbs={breadcrumbs}
+        Footer={<AddRolesScreenFooter onCancel={cancel} />}
+      >
+        <div>
+          <FormTextField
+            required={true}
+            fieldName="key"
+            parentJSONPointer=""
+            type="text"
+            label={renderToString("AddRolesScreen.roleKey.title")}
+            value={formState.roleKey}
+            onChange={onFormStateChangeCallbacks.roleKey}
+          />
+          <WidgetDescription className="mt-2">
+            <FormattedMessage id="AddRolesScreen.roleKey.description" />
+          </WidgetDescription>
+        </div>
+        <div>
+          <FormTextField
+            required={true}
+            fieldName="name"
+            parentJSONPointer=""
+            type="text"
+            label={renderToString("AddRolesScreen.roleName.title")}
+            value={formState.roleName}
+            onChange={onFormStateChangeCallbacks.roleName}
+          />
+          <WidgetDescription className="mt-2">
+            <FormattedMessage id="AddRolesScreen.roleName.description" />
+          </WidgetDescription>
+        </div>
```

Comment:
should be `name` go first

https://linear.app/authgear/issue/DEV-404/roles-and-groups-configuration-in-portal#comment-e5e223c7